### PR TITLE
Introducing Profile Name Parameter and Dockerfile Improvement

### PR DESCRIPTION
**Pull request by cferreirasuazo**
Dear w4af Community,

I am excited to present my latest contribution to the w4af project that brings enhanced usability and eliminates a warning in our Dockerfile. This PR introduces a new feature that allows users to pass a profile name as a parameter to the API, streamlining the workflow and enhancing flexibility. Additionally, I've addressed the TRACEMALLOC warning by adding PYTHONTRACEMALLOC in the Dockerfile.

**Changes Included in this PR:**

1. **Profile Name Parameter Addition:**

    I've implemented a way to pass a profile name as a parameter to the API, eliminating the need to pass the content of the profile name. This enhancement simplifies the API usage and provides users with a more intuitive and user-friendly experience.

2. **Dockerfile Improvement:**

    I've resolved the TRACEMALLOC warning issue by adding the PYTHONTRACEMALLOC environment variable to the Dockerfile. This improvement ensures that w4af runs smoothly in Docker containers without generating unnecessary warnings.

**How to Use the Profile Name Parameter:**

I implemented a 2 steps validation. First, check if the parameter of the scan_profile is a file name, second validate if the profile file exists, then the module does the rest in processing the scanning.

 curl -k -H "Content-Type: application/json" --location 'https://localhost:5000/scans/' \   
--data '{
    "target_urls": ["your_target_url"],
    "scan_profile": "OWASP_TOP10.pw4af"
}

your feedback is welcome in order to improve this PR and next improvements.

### CODE REVIEW

**Comment by cferreirasuazo** [comment]
Hello @security-companion , I was working on a support branch suggested by one of the contributors of the project. I will close this PR and create a new one using my changes and omitting the dependencies check changes.

**Comment by cferreirasuazo** [comment]
Hello @security-companion , 

Thanks for the feedback. 

When learning about the project, I had some trouble running it because of some dependencies., so I visited the issues page of the project in order to find a solution . So I found this [conversation](https://github.com/w4af/w4af/issues/587)  and this [changes](https://github.com/codders/w4af/tree/feat/w4af_api-support]). and the allowed me to run the project.

**Comment by security-companion** [comment]
Hi,
I made a commit on the main branch and the same test failed. So it seems that failure was not caused by your changes.
@codders
Do you know why the test fails?
Are you still planning to merge the changes from https://github.com/codders/w4af/tree/feat/w4af_api-support into this repo?
Greetings

**Comment by codders** [comment]
Hi @security-companion,

I've created #623 to address the failing test-case. Looks like a change in the output from Retirejs.

File: w4af/core/ui/api/utils/scans.py [hunk 1]
```diff
@@ -47,6 +60,13 @@ def create_temp_profile(scan_profile):
     with open(scan_profile_file, 'w') as profile_fh:
         profile_fh.write(scan_profile)
 
+    
+
+    print("***************create_temp_profile*********************")
```
Comment (codders) [inline w4af/core/ui/api/utils/scans.py:65]: Please don't put `print` statements in production code. It's fine if you need them for debugging, but anything that should go to the user should be directed through the `OutputManager` (`om.out.debug(msg)`).

File: w4af/plugins/tests/grep/test_retirejs.py [hunk 2]
```diff
@@ -86,6 +86,7 @@ def test_is_vulnerable_not_detected(self):
  - jQuery before 3.4.0, as used in Drupal, Backdrop CMS, and other products, mishandles jQuery.extend(true, {}, ...) because of Object.prototype pollution
  - Regex in its jQuery.htmlPrefilter sometimes may introduce XSS
  - Regex in its jQuery.htmlPrefilter sometimes may introduce XSS
+ - jQuery 1.x and 2.x are End-of-Life and no longer receiving security updates
```
Comment (codders) [inline w4af/plugins/tests/grep/test_retirejs.py:89]: My pull request #623 hopefully addresses this issue, so you don't need to make this change. As @security-companion says, it's helpful if pull requests address one issue at a time (and not multiple issues in the same PR).

File: extras/docker/Dockerfile [hunk 3]
```diff
@@ -1,10 +1,11 @@
-FROM python:3.10-slim as base
+FROM python:3.11-slim as base
```
Comment (codders) [inline extras/docker/Dockerfile:1]: I would also agree with @security-companion here that the Docker and `dependency_check` changes belong in a separate PR.

**Review by codders — CHANGES_REQUESTED** [review]
Thanks for the great feature add - this looks like it will make life a lot easier for API users!

**Comment by cferreirasuazo** [comment]
as suggested by @codders, I will create multiple PRs. I will close this one. and Send a PR with the Docker update and my change with the profile file.

**Comment by codders** [comment]
Hi @cferreirasuazo,

I see you closed this, but I haven't yet seen the new PR for the feature. We would love to have the profile name parameter feature integrated, and that should resolve #592.

If you need help with making the new / clean PR, please let me know!

**Comment by cferreirasuazo** [comment]
@codders hey, here is the new PR

**Pull request closed without merging**